join.spec.md

```diff
@@ -100,204 +101,349 @@ flowchart TB
+{ fieldB { anywhere } }
+# Generated subgraph operations
+## On B
+{ fieldB { anywhere } }
 ```
 
-// TODO: rename portal query?
-Keys will be passed as `representations` within a [portal query](#portal-query) to [port](#portability) a selection set between subgraphs.
+## Owned fields on owned types
 
-:::[example](photos.graphql#Image) -- Using {@join__type} to specify subgraph keys
+We've finally reached the most interesting case: a field that must be resolved by an operation on a different subgraph from the subgraph on which its parent field was resolved. In order to do this, we need a way to tell the subgraph to resolve that parent object. We do this by defining a special root field in the subgraph's schema: `Query._entities(representations: [_Any!]!): [_Entity]!`. This field takes a list of "representations", which are JSON objects containing a `__typename` field and some other fields, and returns a list of the same length of the corresponding objects resulting from looking up the representations in an application-dependent way.
 
-Multiple {@join}s can be specified for different subgraphs. It is an error to {@join} an object against the same subgraph multiple times.
+There are several ways that the router can calculate a representation to pass to a subgraph. In this specification, all non-value types have a specific subgraph referred to as its "owner", specified via a `@join__owner` directive on the type. (Object types that are not value types are referred to as "entities"; the type `_Entity` referenced above is a union defined in each subgraph's schema consisting of the entity types defined by that subgraph. Only subgraphs which define entities need to define the `Query._entities` field.) The type must also have at least one `@join__type` directive specifying the owning subgraph along with a {key}. For each additional subgraph which can resolve fields returning that type, there should be exactly one `@join__type` directive specifying that subgraph along with a {key}, which should be identical to one of the keys specified with the owning subgraph.
 
-##! @join__owner
+A key is a set of fields on the type (potentially including sub-selections and inline fragments), specified as a string. If a type `T` is annotated with `@join__type(subgraph: G, key: "a b { c }")`, then it must be possible to resolve the full field set provided as a key on subgraph G. Additionally, if you take an object with the structure returned by resolving that field set and add a field `__typename: "T"`, then you should be able to pass the resulting value as a representation to the `Query._entities` field on subgraph G.
 
-Specify the graph which owns the object type.
+In order to resolve a field on an entity on the subgraph that owns its parent type, where that subgraph is different from the subgraph that resolved its parent object, the router first resolves a key for that object on the previous subgraph, and then uses that representation on the owning subgraph.
```

Comment:
This sentence is really difficult to reason about. I'm trying to keep track of a field, an entity, two subgraphs, and a parent type and how they all relate to one another in my head. I think you're explaining the condition which results in needing an `entities` fetch, is that right?

I'm having a hard time putting this into words myself, can we illustrate by example instead?
"Consider the following arrangement..."
"Consider this operation, and specifically this nested field, ..."
"Notice the subgraph ownership changes here, which means we need to perform an entities query"

Or at least getting this into bite size things that will make it simpler for me to visualize crossing the boundary and the arrangement it takes to get to that condition.

Reply:
I'm going to come back to this one because I'm running low on time and it's hard